### ISSUE TITLE
[DevResearch] Button Spinning

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/spark-design-system#3569]
**User Story:**
As Spark we want to Research ...

**AC:**
- a11y with spinner alone
- and a button with a spinner inside the button

**Comment by RVMendoza** [comment]
## A11Y with Spinner Alone
- `aria-label="Loading"` and `role="progressbar" aria-valuetext="Loading"` have the same screen reader outcome
- Spinner does technically represent progress, which is more semantic to give it a `role="progressbar". It needs `aria-valuetext` set because progress is not represented as a number.
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuetext_attribute

### Proposed Spinner State
```html
<div class="sprk-c-Spinner sprk-c-Spinner--circle" role="progressbar" aria-valuetext="Loading"></div>
```



## Button with Spinner
- A loading button should technically be disabled. The user should not be able to interact with it
  - On loading state, we cannot semantically set `disabled` because assistive technology no longer reads the statii nested inside. To prevent that, we should create an additional class that prevents user interaction.
- What happens to our buttons when Finally when all the loading is done? Does it go back to its text? Are there any moments where the button is no longer interactable, and the user could only press it once.
- do not use `aria-busy` some screen readers skip it
- https://accessabilly.com/an-accessible-microinteractions-button/

--------
### Proposed Button

```html
<!--
Default Button
- Note the aria-live="off" addition that now needs to happen by default
-->
<button
  class="sprk-c-Button"
  aria-live="off"
>
  Button Text
</button>


<!--
Loading State Button
- Button needs style attributes that prevent user interaction
-  
-->
<button
  class="sprk-c-Button sprk-c-Button--loading"
  data-sprk-spinner="click"
  aria-live="assertive"
>
  <div class="sprk-c-Spinner sprk-c-Spinner--circle"  role="progressbar" aria-valuetext="Loading"></div>
</button>

```

-------
### Polaris is going through something similar
https://github.com/Shopify/polaris-react/issues/1212
> The main problem with this implementation is that it requires having aria-live="off" on the button component by default. It feels like overkill, but we don't know when a consumer will end up making a button go into a loading state, and we don't want button content to be read on page load. 🤷‍♀
